scrapegraphai/nodes/generate_answer_node.py

```diff
@@ -99,44 +100,48 @@ def execute(self, state: dict) -> dict:
         for i, chunk in enumerate(tqdm(doc, desc="Processing chunks", disable=not self.verbose)):
             if len(doc) == 1:
+                # No batching needed for single chunk
                 prompt = PromptTemplate(
-                    template=template_no_chunks_prompt,
+                    template=template_no_chunks,
                     input_variables=["question"],
                     partial_variables={"context": chunk.page_content,
-                                       "format_instructions": format_instructions})
-                chain =  prompt | self.llm_model | output_parser
+                                    "format_instructions": format_instructions})
+                chain = prompt | self.llm_model | output_parser
                 answer = chain.invoke({"question": user_prompt})
-
+                
             else:
+                # Prepare prompt with chunk information
                 prompt = PromptTemplate(
-                    template=template_chunks_prompt,
+                    template=template_chunks,
                     input_variables=["question"],
                     partial_variables={"context": chunk.page_content,
-                                        "chunk_id": i + 1,
-                                        "format_instructions": format_instructions})
-
-            # Dynamically name the chains based on their index
-            chain_name = f"chunk{i+1}"
-            chains_dict[chain_name] = prompt | self.llm_model | output_parser
+                                    "chunk_id": i + 1,
+                                    "format_instructions": format_instructions})
+                # Add chain to dictionary with dynamic name
+                chain_name = f"chunk{i+1}"
+                chains_dict[chain_name] = prompt | self.llm_model | output_parser
 
+            # Batch process chains if there are multiple chunks
         if len(chains_dict) > 1:
-            # Use dictionary unpacking to pass the dynamically named chains to RunnableParallel
-            map_chain = RunnableParallel(**chains_dict)
-            # Chain
-            answer = map_chain.invoke({"question": user_prompt})
-            # Merge the answers from the chunks
-            merge_prompt = PromptTemplate(
-                template = template_merge_prompt,
-                input_variables=["context", "question"],
-                partial_variables={"format_instructions": format_instructions},
-            )
-            merge_chain = merge_prompt | self.llm_model | output_parser
-            answer = merge_chain.invoke({"context": answer, "question": user_prompt})
-
-        # Update the state with the generated answer
-        state.update({self.output[0]: answer})
+            async def process_chains():
+                async_runner = AsyncRunnable()
+                for chain_name, chain in chains_dict.items():
+                    async_runner.add(chain.abatch([{"question": user_prompt}] * len(doc)))
```

Comment:
why would you pass the same argument `user_prompt` in a batch of size the document length, without varying the chunk's context?

should use the chain's `ainvoke` method